teos/src/tx_index.rs

```diff
@@ -248,7 +265,10 @@ mod tests {
         let last_six_blocks = get_last_n_blocks(&mut chain, 6).await;
         let blocks: Vec<Block> = last_six_blocks
             .iter()
-            .map(|block| block.deref().clone())
+            .map(|block| match block.deref() {
+                lightning_block_sync::BlockData::FullBlock(b) => b.clone(),
+                _ => panic!("Expected FullBlock"),
+            })
             .collect();
```

Comment:
let's have utility function for this on top of the module before the first test.

```rust
let last_six_blocks = get_last_n_blocks(&mut chain, 6).await;
let full_blocks = get_full_blocks(&last_six_blocks); // type: Vec<Block>
```